aws/eks-anywhere
# Support linking VM resources with Projects on Nutanix

These changes allow specifying an optional project in the NutanixMachineConfig
which, when specified, is propagated to the corresponding NutanixMachineTemplate
property. 

**How has this been tested?**
- Used a bundle-release.yaml with eksa-components.yaml and cluster-controller image overridden to
the ones corresponding to the changes in this PR.
- Set the `project` property in `NutanixMachineConfig` manifests to `my-project`: A project that already exists in Prism Central
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: NutanixMachineConfig
metadata:
  name: eksa-sid-cp
spec:
  ...
  project:
    name: my-project
    type: name
  ...

---
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: NutanixMachineConfig
metadata:
  name: eksa-sid
spec:
  ...
  project:
    name: my-project
    type: name
  ...
```
- Created a cluster successfully
```
$ bin/eksctl-anywhere create cluster -f ./${CLUSTER_NAME}.yaml --force-cleanup -v 10  --bundles-override ./bundle-release-project.yaml
...
2023-03-15T23:00:00.450+0100	V0	🎉 Cluster created!
```
- Check the names of the machines in the created cluster
```
$ kubectl get machines -A
NAMESPACE     NAME                             CLUSTER    NODENAME                         PROVIDERID                                       PHASE     AGE   VERSION
eksa-system   eksa-sid-22rsp                   eksa-sid   eksa-sid-22rsp                   nutanix://33182ce4-e87e-4145-93f5-9441157c6e00   Running   21m   v1.24.10-eks-1-24-12
eksa-system   eksa-sid-476bp                   eksa-sid   eksa-sid-476bp                   nutanix://d82e1a2e-7c65-4f90-bfe8-5d3a28934e21   Running   21m   v1.24.10-eks-1-24-12
eksa-system   eksa-sid-bkvlk                   eksa-sid   eksa-sid-bkvlk                   nutanix://62754b9a-258b-4efb-bf2c-ff9d8de880c0   Running   21m   v1.24.10-eks-1-24-12
eksa-system   eksa-sid-md-0-85b84bd54d-2l4gb   eksa-sid   eksa-sid-md-0-85b84bd54d-2l4gb   nutanix://47989010-bd02-45be-abf0-4fd5fd3c668c   Running   21m   v1.24.10-eks-1-24-12
eksa-system   eksa-sid-md-0-85b84bd54d-hx9st   eksa-sid   eksa-sid-md-0-85b84bd54d-hx9st   nutanix://69494499-4c38-48a4-bc92-9aff8d89ca8c   Running   21m   v1.24.10-eks-1-24-12
eksa-system   eksa-sid-md-0-85b84bd54d-rsndq   eksa-sid   eksa-sid-md-0-85b84bd54d-rsndq   nutanix://0aa24440-e038-4ab8-8b55-df3900aec941   Running   21m   v1.24.10-eks-1-24-12
```
- Verify on Prism Central that the VMs corresponding to the machines above are linked to the `my-project` project.
![2023-03-15 at 23 06](https://user-images.githubusercontent.com/6081171/225456613-696bd320-f2cb-47c7-99d5-cd37cae5d407.png)


Full install logs: https://app.warp.dev/block/ZQ5t4rdhmUL0GCNQZ2JqNW
